WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift

```diff
@@ -83,15 +83,122 @@ extension PublishingEditor {
         }
     }
 
-    func handlePublishButtonTap() {
+    func handlePrimaryActionButtonTap() {
         let action = self.postEditorStateContext.action
 
-        publishPost(
-            action: action,
-            dismissWhenDone: action.dismissesEditor,
-            analyticsStat: self.postEditorStateContext.publishActionAnalyticsStat)
+        guard RemoteFeatureFlag.syncPublishing.enabled() else {
+            publishPost(
+                action: action,
+                dismissWhenDone: action.dismissesEditor,
+                analyticsStat: self.postEditorStateContext.publishActionAnalyticsStat)
+            return
+        }
+
+        performEditorAction(action, analyticsStat: postEditorStateContext.publishActionAnalyticsStat)
+    }
+
+    func handleSecondaryActionButtonTap() {
+        guard let action = self.postEditorStateContext.secondaryPublishButtonAction else {
+            // If the user tapped on the secondary publish action button, it means we should have a secondary publish action.
+            let error = NSError(domain: EditorError.errorDomain, code: EditorError.expectedSecondaryAction.rawValue, userInfo: nil)
+            WordPressAppDelegate.crashLogging?.logError(error)
+            return
+        }
+
+        let secondaryStat = self.postEditorStateContext.secondaryPublishActionAnalyticsStat
+
+        let publishPostClosure = { [unowned self] in
+            guard RemoteFeatureFlag.syncPublishing.enabled() else {
+                publishPost(action: action, dismissWhenDone: action.dismissesEditor, analyticsStat: secondaryStat)
+                return
+            }
+            performEditorAction(action, analyticsStat: secondaryStat)
+        }
+
+        if presentedViewController != nil {
+            dismiss(animated: true, completion: publishPostClosure)
+        } else {
+            publishPostClosure()
+        }
+    }
+
+    func performEditorAction(_ action: PostEditorAction, analyticsStat: WPAnalyticsStat?) {
```

Comment:
This change should probably not be part of this PR, but it's too late too change. I suggest we roll forward and continue working on it in the future PRs. It's in the branch ,so it's safe.